File: SESION2/sesion2/src/meetnmatch/FactoriaTenis.java
package meetnmatch;

import java.util.*;

public class FactoriaTenis implements Factoria {

    @Override
	public EncuentroTenis crearPartido() {
		EncuentroTenis enc=new EncuentroTenis("Polideportivo",1.2);

		enc.apuntarseLocal(new JugadorTenis("John",15,"11111111A","Zurdo",1.78));
		enc.apuntarseVisitante(new JugadorTenis("Michael",16,"11112115A", "Diestro",1.7));
                
                return enc;

	}
}

File: SESION2/ruby/MeetNMatch/FactoriaTenis.rb
require 'JugadorTenis'
require 'EncuentroTenis'
require 'Encuentro'
require 'Factoria'

module MeetNMatch
	
	class FactoriaTenis < Factoria

		@PrototipoJugadorTenis
		@PrototipoEncuentroTenis

		def crearPartido(local)
			encuentro=@PrototipoEncuentroTenis.clone
			
			encuentro.localizacion=local
			
			encuentro.apuntarseLocal(@PrototipoJugadorTenis.clone)

			encuentro.apuntarseVisitante(@PrototipoJugadorTenis.clone)

			return encuentro		
		end

		def initialize
			@PrototipoEncuentroTenis=EncuentroTenis.new(1)
			@PrototipoJugadorTenis=JugadorTenis.new("Diestro",1)
		end
	end
end
